store/types/store.go

```diff
@@ -194,9 +194,11 @@ type CommitMultiStore interface {
 	// ListeningEnabled returns if listening is enabled for the KVStore belonging the provided StoreKey
 	ListeningEnabled(key StoreKey) bool
 
-	// AddListeners adds WriteListeners for the KVStore belonging to the provided StoreKey
-	// It appends the listeners to a current set, if one already exists
-	AddListeners(key StoreKey, listeners []WriteListener)
+	// AddListeners adds q listener for the KVStore belonging to the provided StoreKey
+	AddListeners(keys []StoreKey)
+
+	// PopStateCache returns the accumulated state change messages from MemoryListener
+	PopStateCache() []*StoreKVPair
```

Comment:
1. It returns the state changes messages accumulated from the _CommitMultiStore_ which the method is implemented by
2. It _destroys_ those state change messages in its implementation, which is a mutating operation, that isn't described in documentation or synchronized in any way

Reply:
added comments to implementation and synchronized method.